Source/ACE/Entity/Player.cs

```diff
@@ -194,6 +194,43 @@ public void HandleActionLearnSpell(uint spellId)
             learnSpellChain.EnqueueChain();
         }
 
+        /// <summary>
+        /// This method implements player spell bar management for - adding a spell to a specific spell bar (0 based) at a specific slot (0 based).
+        /// </summary>
+        /// <param name="spellId"></param>
+        /// <param name="spellBarPositionId"></param>
+        /// <param name="spellBarId"></param>
+        public void HandleActionAddSpellToSpellBar(uint spellId, uint spellBarPositionId, uint spellBarId)
+        {
+            // The spell bar magic happens here. First, let's mind our race conditions....
+            // No need for action chain here - internal to us, no other object can change us.
+            AceObject.SpellsInSpellBars.Add(new AceObjectPropertiesSpellBarPositions()
+                                                { AceObjectId = AceObject.AceObjectId,
+                                                  SpellId = spellId,
+                                                  SpellBarId = spellBarId,
+                                                  SpellBarPositionId = spellBarPositionId });
+        }
+
+        /// <summary>
+        /// This method implements player spell bar management for - removing a spell to a specific spell bar (0 based)
+        /// </summary>
+        /// <param name="spellId"></param>
+        /// <param name="spellBarId"></param>
+        public void HandleActionRemoveSpellToSpellBar(uint spellId, uint spellBarId)
+        {
+            // More spell bar magic happens here. First, let's mind our race conditions....
+            // No need for action chain here - internal to us, no other object can change us.
+            AceObject.SpellsInSpellBars.Remove(AceObject.SpellsInSpellBars.Single(x => x.SpellBarId == spellBarId && x.SpellId == spellId));
+            // Now I have to reorder
+            var sorted = AceObject.SpellsInSpellBars.FindAll(x => x.AceObjectId == AceObject.AceObjectId && x.SpellBarId == spellBarId).OrderBy(s => s.SpellBarPositionId);
```

Comment:
In general I don't like exposing the properties of AceObject outside of the AceObject (it requires the user of AceObject to know how the data is stored, which shouldn't be exposed to the game-content data).  Can you write an accessor for this (and other accesses of this type)?